src/PostgREST/App.hs

```diff
@@ -206,7 +204,9 @@ app dbStructure conf apiRequest =
         else return notFound
 
     (ActionRead, TargetRoot, Nothing) -> do
-      body <- encode <$> H.query schema accessibleTables
+      body <- if contentType == OpenAPI
```

Comment:
can be a bit more clean
```
body <- encodeFn <$> H.query schema accessibleTables
where encodeFn = if contentType == OpenAPI then encodeApi . toTableInfo else encode
```
and maybe also the encodeFn definition can embed these lines
```
encodeApi ti = encodeOpenAPI ti host port
host = configHost conf
port = toInteger $ configPort conf
```